content/SDKs/javascript/api-reference.mdx

```diff
@@ -260,6 +263,143 @@ const deletedCount = await context.vector.delete('product-descriptions', 'id1');
 console.log(`Deleted ${deletedCount} vector(s)`);
 ```
 
+### Object Storage
+
+The Object Storage API provides a way to store and retrieve objects (files, documents, media) with support for public URL generation. It is accessed through the `context.objectstore` object.
+
+#### `get(bucket: string, key: string): Promise<DataResult>`
+
+Retrieves an object from the object storage.
+
+##### Parameters
+
+- `bucket`: The bucket to get the object from
+- `key`: The key of the object to get
+
+##### Return Value
+
+Returns a Promise that resolves to a DataResult object. The DataResult has an `exists` boolean property and a `data` property containing the object data if found.
+
+##### Example
+
+```typescript
+// Retrieve an object from object storage
+const result = await context.objectstore.get('user-uploads', 'profile-123.jpg');
+if (result.exists) {
+  // Access data using the Data interface methods
+  const imageData = await result.data.binary();
+  console.log(`Image size: ${imageData.byteLength} bytes`);
+} else {
+  console.log('Image not found');
+}
+```
+
+#### `put(bucket: string, key: string, data: DataType, params?: ObjectStorePutParams): Promise<void>`
+
+Stores an object in the object storage.
+
+##### Parameters
```

Comment:
we should document that the contentType parameter is optional and if not provided will attempt to set based on the data type passed. for example, if an object is passed the contenttype will be assumed to be json.

if we cannot determine the contentType we will set to application/octet-stream